### PR TITLE
Test to verify diag_encl

### PR DESCRIPTION
**Pull request by Pavithra1602**
This test verifies '-d' option of diag_encl. It checks whether single xml file is generated when "diag_encl" command is run multiple times and validates xml content.

Signed-off-by: Pavithra <pavrampu@linux.vnet.ibm.com>

### CODE REVIEW

File: generic/diag_encl.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        self.log.info("===========Executing diag_encl test==========")
+        diag_path = '/var/log/ppc64-diag/diag_disk/'
+        if not os.path.isdir(diag_path):
+            self.fail('diag_disk path does not exists.')
+        if not self.run_cmd_out("diag_encl -h | grep -Eai '\-d'").strip():
+            self.fail("'-d' option is not available in help message")
+        for i in range(4):
+            self.run_cmd("diag_encl -d")
+        xml_file_path = os.path.join(diag_path, '*diskAnalytics*')
+        no_of_files = self.run_cmd_out("ls -lrt %s | wc -l" % xml_file_path).strip()
```
Comment (narasimhan-v) [inline generic/diag_encl.py]: Better to use python libraries to do this

File: generic/diag_encl.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        if no_of_files == '0':
+            self.fail("xml file not generated")
+        if no_of_files > '1':
+            self.fail("multiple xml files are generated")
+        else:
```
Comment (narasimhan-v) [inline generic/diag_encl.py]: else is redundant here

File: generic/diag_encl.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        machine_type = e.find('Machine').get('type').strip()
+        machine_model = e.find('Machine').get('model').strip()
+        machine_serial_xml = e.find('Machine').get('serial').strip()
+        if 'KVM' in self.run_cmd_out("pseries_platform"):
+            product_path = '/proc/device-tree/host-model'
+            serial_path = '/proc/device-tree/host-serial'
+        else:
+            product_path = '/proc/device-tree/model'
+            serial_path = '/proc/device-tree/system-id'
+        product_name = self.run_cmd_out("cat %s" % product_path).rstrip(' \t\r\n\0')
```
Comment (narasimhan-v) [inline generic/diag_encl.py]: please use `genio.read_file()`

**Comment by Pavithra1602** [comment]
opened new PR

**Pull request closed without merging**